clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1672,6 +1682,11 @@ private TxnOffsetCommitHandler(TransactionalRequestResult result,
             this.builder = builder;
         }
 
+        private TxnOffsetCommitHandler(TxnOffsetCommitRequest.Builder builder) {
+            super("TxnOffsetCommitHandler");
+            this.builder = builder;
```

Comment:
In the above, we pass the result to super. What does it mean to pass the name of the class here?

Reply:
It creates a sort of future variable `result` with this name string.

Reply:
There are a few other examples in this file
```
private AddPartitionsToTxnHandler(AddPartitionsToTxnRequest.Builder builder) {
            super("AddPartitionsToTxn");
            this.builder = builder;
            this.retryBackoffMs = TransactionManager.this.retryBackoffMs;
        }

private FindCoordinatorHandler(FindCoordinatorRequest.Builder builder) {
            super("FindCoordinator");
            this.builder = builder;
        }

```

Reply:
Oh interesting. I think I was more wondering what the other TxnOffsetCommitHandler string looks like.

Reply:
Taking a look, it seems like we pass a result if it exists, otherwise the string param creates a new result with that string name. Seems like this should be ok 👍